deployer/grafana/grafana_tokens.py

```diff
@@ -0,0 +1,208 @@
+import json
+from base64 import b64encode
+from pathlib import Path
+
+import requests
+import typer
+
+from ..cli_app import app
+from ..utils import print_colour
+from .grafana_utils import (
+    get_grafana_admin_password,
+    get_grafana_url,
+    update_central_grafana_token,
+)
+
+REPO_ROOT = Path(__file__).parent.parent.parent
+
+
+def build_service_account_request_headers():
+    """
+    Build the headers needed to send requests to the Grafana service account endpoint.
+
+    Returns:
+        dict: "Accept", "Content-Type", "Authorization" headers
+    """
+    # Get grafana credentials
+    grafana_username = "admin"
+    grafana_password = get_grafana_admin_password()
+    credentials = f"{grafana_username}:{grafana_password}"
+    basic_auth_credentials = b64encode(credentials.encode("utf-8"))
+
+    # Build and return the headers
+    return {
+        "Accept": "application/json",
+        "Content-Type": "application/json",
+        "Authorization": "Basic " + basic_auth_credentials.decode("utf-8"),
+    }
+
+
+def create_deployer_service_account(sa_endpoint, headers):
+    """
+    Create a Grafana service account with the name: `deployer`
+
+    Returns:
+        id: the id of the service account with the name `deployer`
+    """
+
+    request_body = {"name": "deployer", "role": "Admin", "isDisabled": False}
+
+    # Create a service account called `deployer` in the cluster's grafana
+    response = requests.post(
+        sa_endpoint,
+        data=json.dumps(request_body),
+        headers=headers,
+    )
+    if not response.ok:
+        print(
+            "An error occured while creating the service account for the deployer.\n"
+            f"Error was {response.text}."
+        )
+        response.raise_for_status()
+    print_colour("Successfully created the `deployer` sa!")
+
+    return response.json()["id"]
+
+
+def get_deployer_service_account_id(sa_endpoint, headers):
+    """
+    Iterate the list of service accounts and
+    retrieve the id of the service account which has the name "deployer".
+
+    Assumes:
+        the maximum number of service accounts created for a grafana is no more than 10.
+
+    Returns:
+        id: the id of the service account with the name `deployer`
+            `None` if no service account with this name was found
+    """
+
+    # Assumes we don't have more than 10 service accounts
+    # FIXME if this changes
```

Comment:
We already have 17 clusters - 18 once #2022 is merged. Assuming one grafana instance per cluster and they will all be using a service account, maybe best to bump this number now?

Reply:
Hmmm, I guess there are some long-lived tokens that will mean we might not get to this number?

Reply:
*` # Assumes we don't have more than 10 service accounts`* should have been *` # Assumes we don't have more than 10 service accounts per Grafana`*.

Sorry for the confusion. I added a commit to improve the comments around the service accounts per Grafana https://github.com/2i2c-org/infrastructure/pull/1938/commits/7645df1e66a1ba32a0bca6e82d3f8e379407700b.
 LMK is the comments are more clear now.

Basically, with this PR, we are creating the infrastructure to create our first service accounts **per Grafana**, called `deployer`. In case we will create other service accounts in the future for rbac purposes for example, and that number is >10 we should update this. But my guess is that this might take a while, if ever.

Reply:
Ahhh, per grafana makes way more sense - thanks!